scripts/metadata_table.py

```diff
@@ -813,8 +858,11 @@ def __init_from_file(self, table_name, table_type, known_ddts, run_env, skip_ddt
         if self.header_type == "ddt":
             known_ddts.append(self.title)
         # end if
-        # We need a default module if none was listed
-        if self.module is None:
+        # We need a default module if none was listed.
+        # DJS2024: First, try to find module_name from the metadata. Otherwise,
+        # use file name as module_name (default).
+        self.__module_name = find_module_name(self.__pobj.filename)
+        if (self.__module_name == ''):
```

Comment:
How can this happen? We require all Fortran code to be in modules, schemes or not?

Reply:
The metadata is only needed for the DDT, not the module where it lives. 
The behavior now is that IF you provide a module table entry in the DDT metadata file it will use that instead.